test/float8/test_fsdp2/test_fsdp2.py

```diff
@@ -59,7 +59,7 @@ def init_multi_module(self) -> nn.Module:
         self.broadcast_module(module)
         return module
 
-    def init_transformer(self, weight_tying: bool) -> nn.Module:
+    def init_transformer(self, weight_tying: bool, dtype: torch.dtype | None = None) -> nn.Module:
```

Comment:
seems CI complained the grammar for python 3.9, `torch.dtype | None = None`

maybe `Optional[torch.dtype] = None` ?